open-apparel-registry/open-apparel-registry
# Add ExtendedFields model

## Overview

Adds the ExtendedFields model, the first step towards extended profiles.

The `field_name`, describing the field name for the extended field,
is pulled from a list of choices. Currently, available field_names are
country, address, name, number_of_workers and native_language_name.

The `value` is stored in a jsonb object, with different structure for
different field types. Numeric fields will be stored as {“min”: v1,
“max”: v2}. If there is a single numeric value, we will set both min
and max to it. If there is no provided value, we won't make an
ExtendedField row.

Each ExtendedField will be created from either a facility claim or a
facility list item. It will be must linked to one or the other, but
not both.

Connects #1492 

## Demo

<img width="950" alt="Screen Shot 2021-10-26 at 2 16 20 PM" src="https://user-images.githubusercontent.com/21046714/138940649-c8f01912-cc13-40e7-8285-8314e0abba97.png">

## Testing Instructions

* Run `./scripts/manage migrate` and confirm the migration is successful.
* Run `./scripts/manage shell_plus` and create an ExtendedField (replace the facility and list_item ids with ones from your database).
```
value = { "min": 100, "max": 500 }
ExtendedField.objects.create(contributor_id=3, facility_id='CN2021299G038W1', facility_list_item_id='372', field_name='number_of_workers', value=value)
```
* Confirm that the Extended field was created successfully and has the correct data content.  
```
ExtendedField.objects.get(id=1).__dict__
```
## Checklist
